### DISCUSSION TITLE
What is the simplest way to convert an Array column to and from numpy \ pytorch \ tensorflow?

### DISCUSSION THREAD

**Discussion by ophiry** [ibis-project/ibis#9287] [Q&A] [answered]
sklearn, pytorch, tensorflow rely mostly on matrix input and outputs, what would be the simplest (and preferably efficient) way to integrate them with ibis?

is there also support for sparse vectors?

**Comment by gforsyth** [comment, marked as answer]
Hi @ophiry -- we don't have support for sparse vectors.

For `pytorch`, we have a `to_torch()` method on tables and columns for generating a torch-compatible version of the table/column.

For `numpy`, the fastest route is via `pyarrow`, so you could call `my_column.to_pyarrow().to_numpy()`.

I haven't used `tensorflow` in a loooooong time, is there a particular input format it prefers?
